cmd2/command_definition.py

```diff
@@ -126,3 +133,36 @@ def on_unregistered(self) -> None:
         Subclasses can override this to perform remaining cleanup steps.
         """
         self._cmd = None
+
+    @property
+    def settable_prefix(self) -> str:
+        return self._settable_prefix
+
+    @property
+    def settables(self) -> Mapping[str, Settable]:
+        return self._settables
+
+    def add_settable(self, settable: Settable) -> None:
+        """
+        Convenience method to add a settable parameter to the CommandSet
+
+        :param settable: Settable object being added
+        """
+        if self._cmd and not self._cmd.always_prefix_settables:
+            if settable.name in self._cmd.settables.keys() and settable.name not in self._settables.keys():
+                raise KeyError(f'Duplicate settable: {settable.name}')
+        if settable.settable_obj is None:
+            settable.settable_obj = self
+        self._settables[settable.name] = settable
```

Comment:
Does this consider the case where two CommandSets used the same prefix and the same setting name?

Reply:
No. We can either add a check to prevent 2 commandsets from using the same prefix (easy) or do more complex checking.